RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelRawToClusterCUDA.cc

```diff
@@ -83,19 +86,29 @@ class SiPixelRawToClusterCUDA : public edm::stream::EDProducer<edm::ExternalWork
 };
 
 SiPixelRawToClusterCUDA::SiPixelRawToClusterCUDA(const edm::ParameterSet& iConfig)
-    : rawGetToken_(consumes<FEDRawDataCollection>(iConfig.getParameter<edm::InputTag>("InputLabel"))),
-      digiPutToken_(produces<cms::cuda::Product<SiPixelDigisCUDA>>()),
+    : digiPutToken_(produces<cms::cuda::Product<SiPixelDigisCUDA>>()),
       clusterPutToken_(produces<cms::cuda::Product<SiPixelClustersCUDA>>()),
-      gpuMapToken_(esConsumes<SiPixelROCsStatusAndMappingWrapper, CkfComponentsRecord>()),
-      gainsToken_(esConsumes<SiPixelGainCalibrationForHLTGPU, SiPixelGainCalibrationForHLTGPURcd>()),
-      cablingMapToken_(esConsumes<SiPixelFedCablingMap, SiPixelFedCablingMapRcd>(
-          edm::ESInputTag("", iConfig.getParameter<std::string>("CablingMapLabel")))),
+      isUpgrade_(iConfig.getParameter<bool>("isUpgrade")),
       isRun2_(iConfig.getParameter<bool>("isRun2")),
       includeErrors_(iConfig.getParameter<bool>("IncludeErrors")),
       useQuality_(iConfig.getParameter<bool>("UseQualityInfo")),
       maxFedWords_(iConfig.getParameter<uint32_t>("MaxFEDWords")),
       clusterThresholds_{iConfig.getParameter<int32_t>("clusterThreshold_layer1"),
                          iConfig.getParameter<int32_t>("clusterThreshold_otherLayers")} {
+  if (isRun2_ and isUpgrade_) {
+    throw cms::Exception("SiPixelRawToClusterCUDA") << "Cannot use isRun2 and isUpgrade at the same time\n";
```

Comment:
```suggestion
    throw cms::Exception("Configuration") << "Cannot use isRun2 and isUpgrade at the same time\n";
```
would be more standard category